cmd/openshift-tuned.go

```diff
@@ -201,24 +205,109 @@ func profilesExtract() error {
 			return fmt.Errorf("Failed to create tuned profile file '%s': %v", profileFile, err)
 		}
 		defer f.Close()
-		_, err = f.WriteString(value)
-		if err != nil {
+		if _, err = f.WriteString(value); err != nil {
 			return fmt.Errorf("Failed to write tuned profile file '%s': %v", profileFile, err)
 		}
 	}
 	return nil
 }
 
+func openshiftTunedPidFileWrite() error {
```

Comment:
What are you using this PID file for?

Reply:
It is used by the pod's `preStop` lifecycle hook to 1) send SIGTERM to openshift-tuned, 2) wait for openshift-tuned to send tuned SIGTERM to do the actual settings rollback.  I tried to do this without writing an additional PID, but failed.  There might be a more elegant solution without writing the PID, unfortunately I don't know a better way at this point that would work for me.  Another option (without writing a new PID file) might be using the PID written by the tuned daemon itself, but that would also require some additional coding and the current code is probably cleaner.